src/spiral_stc.cpp

```diff
@@ -8,294 +8,356 @@
+
+    SpiralSTC::makePlan(start, goal, plan);
+
+    global_path.poses.clear();
+    global_path.header.stamp = node_->now();
+    global_path.header.frame_id = global_frame_;
+    for (auto pose : plan)
     {
-      // Initialize spiral direction towards y-axis
-      dx = 0;
-      dy = 1;
+      global_path.poses.push_back(pose);
     }
-    done = true;
-
-    for (int i = 0; i < 4; ++i)
+    return global_path;
+  }
+  std::list<gridNode_t> SpiralSTC::spiral(std::vector<std::vector<bool>> const &grid, std::list<gridNode_t> &init,
+                                          std::vector<std::vector<bool>> &visited)
+  {
+    int dx, dy, dx_prev, x2, y2, nRows = grid.size(), nCols = grid[0].size();
+    // Spiral filling of the open space
+    // Copy incoming list to 'end'
+    std::list<gridNode_t> pathNodes(init);
+    // Create iterator for gridNode_t list and let it point to the last element of end
+    std::list<gridNode_t>::iterator it = --(pathNodes.end());
+    if (pathNodes.size() > 1) // if list is length 1, keep iterator at end
+      it--;                   // Let iterator point to second to last element
+
+    gridNode_t prev = *(it);
+    bool done = false;
+    while (!done)
     {
-      x2 = pathNodes.back().pos.x + dx;
-      y2 = pathNodes.back().pos.y + dy;
-      if (x2 >= 0 && x2 < nCols && y2 >= 0 && y2 < nRows)
+      if (it != pathNodes.begin())
+      {
+        // turn ccw
+        dx = pathNodes.back().pos.x - prev.pos.x;
+        dy = pathNodes.back().pos.y - prev.pos.y;
+        dx_prev = dx;
+        dx = -dy;
+        dy = dx_prev;
+      }
+      else
       {
-        if (grid[y2][x2] == eNodeOpen && visited[y2][x2] == eNodeOpen)
+        // Initialize spiral direction towards y-axis
+        dx = 0;
+        dy = 1;
+      }
+      done = true;
+
+      for (int i = 0; i < 4; ++i)
+      {
+        x2 = pathNodes.back().pos.x + dx;
+        y2 = pathNodes.back().pos.y + dy;
+        if (x2 >= 0 && x2 < nCols && y2 >= 0 && y2 < nRows)
         {
-          Point_t new_point = { x2, y2 };
-          gridNode_t new_node =
+          if (grid[y2][x2] == eNodeOpen && visited[y2][x2] == eNodeOpen)
           {
-            new_point,  // Point: x,y
-            0,          // Cost
-            0,          // Heuristic
-          };
-          prev = pathNodes.back();
-          pathNodes.push_back(new_node);
-          it = --(pathNodes.end());
-          visited[y2][x2] = eNodeVisited;  // Close node
-          done = false;
-          break;
+            Point_t new_point = {x2, y2};
+            gridNode_t new_node =
+                {
+                    new_point, // Point: x,y
+                    0,         // Cost
+                    0,         // Heuristic
+                };
+            prev = pathNodes.back();
+            pathNodes.push_back(new_node);
+            it = --(pathNodes.end());
+            visited[y2][x2] = eNodeVisited; // Close node
+            done = false;
+            break;
```

Comment:
You're jumping out of the loop by `break`. So why have the whole `bool done` bookkeeping?